### PR TITLE
Mobile release automation

### PR DESCRIPTION
**Pull request by tbezman**
It's time to automate our release process. You can read more about this PR in this notion doc https://www.notion.so/gallery-so/Mobile-Release-Management-e7579fb89bb447f5b0a8d2b7ebc39af1

I'll tag up some sections of code in this PR but otherwise I suggest you read the notion doc.

### CODE REVIEW

File: .github/workflows/mobile-release.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+name: mobile-release
```
Comment (tbezman) [inline .github/workflows/mobile-release.yml:1]: only difference between this file and `mobile-build.yml` is the `--auto-submit` flag.

File: apps/mobile/app.json [hunk 2]
```diff
@@ -17,12 +16,13 @@
     },
     "assetBundlePatterns": ["**/*"],
     "ios": {
-      "buildNumber": "5",
+      "config": {
+        "usesNonExemptEncryption": false
+      },
```
Comment (tbezman) [inline apps/mobile/app.json:21]: If I don't add this here, I have to manually accept something in AppStoreConnect. This is totally chill.

File: apps/mobile/eas.json [hunk 3]
```diff
@@ -1,28 +1,33 @@
...
+  "submit": {
+    "development": {
+      "ios": {
+        "ascAppId": "1672735556"
       }
     },
     "production": {
       "ios": {
-        "resourceClass": "m1-medium"
+        "ascAppId": "1672735556"
```
Comment (tbezman) [inline apps/mobile/eas.json:29]: this is chill and not secret.

File: apps/mobile/index.js [hunk 4]
```diff
@@ -1,10 +1,7 @@
 import registerRootComponent from 'expo/build/launch/registerRootComponent';
```
Comment (tbezman) [inline apps/mobile/index.js:1]: The `index` file has to be `.js` or Android builds don't work. This doesn't super bother me but is weird.

**Review by Robinnnnn — APPROVED** [review]
🚀🚀🚀

**Pull request merged**